target/arm/helper.c

```diff
@@ -12726,8 +12726,10 @@ static bool get_phys_addr_lpae(CPUARMState *env, uint64_t address,
 
     // Cap stores can fault here as only tagged stores specify
     // MMU_DATA_CAP_STORE
-    if (!cdbm && !sc) {
-        *prot |= PAGE_SC_TRAP;
+    // Software capability dirty tracking means we fault on !sc
+    if (!sc) {
+        if (!cdbm)
```

Comment:
I don't understand why this needs checking with the way Armv8 defines DBM. This function isn't checking the normal DBM, for example. SC is the "can I store capabilities bit" and CDBM is the "atomically update SC to 1 when you would otherwise fault due to SC=0" bit.

Reply:
This should probably cdbm completely if you want traps because we don't support hardware modification of PTEs. This patch would cause accesses that cause TLB refill to act differently than those that hit the TLB, which is certainly wrong.